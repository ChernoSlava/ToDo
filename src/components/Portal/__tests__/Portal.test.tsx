// eslint-disable import/no-extraneous-dependencies /
// eslint-disable no-undef /

import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { theme } from '@theme';
import { Portal } from '..';

const getUi = (node?: Element): JSX.Element => {
  return (
    <ThemeProvider theme={theme}>
      <Portal domNode={node}>
        test
      </Portal>
    </ThemeProvider>
  );
};

describe('Portal tests', () => {
  it('should render by default', () => {
    const { container } = render(getUi(), { container: document.body });

    expect(container).toMatchSnapshot();
  });

  it('should render in Root', () => {
    const root = document.createElement('div');
    root.setAttribute('id', 'root');
    document.body.append(root);
    const { container } = render(getUi(), { container: root });

    expect(container).toMatchSnapshot();
  });

  it('should render in Custom node', () => {
    const customNode = document.createElement('div');
    document.body.appendChild(customNode);
    
    const { container } = render(getUi(customNode), {
      container: customNode,
    });

    expect(container).toMatchSnapshot();
  });
});
