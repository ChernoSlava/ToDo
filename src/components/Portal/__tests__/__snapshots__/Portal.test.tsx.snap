// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal tests should render by default 1`] = `
<body>
  test
</body>
`;

exports[`Portal tests should render in Custom node 1`] = `
<div>
  test
</div>
`;

exports[`Portal tests should render in Root 1`] = `
<div
  id="root"
>
  test
</div>
`;
